core/store/src/flat/types.rs

```diff
@@ -81,6 +83,12 @@ impl Into<i64> for &FlatStorageStatus {
                 FlatStorageCreationStatus::FetchingState(_) => 11,
                 FlatStorageCreationStatus::CatchingUp(_) => 12,
             },
+            // 20..30 is reserved for resharding statuses.
+            FlatStorageStatus::Resharding(resharding_status) => match resharding_status {
+                FlatStorageReshardingStatus::SplittingParent => 20,
+                FlatStorageReshardingStatus::CreatingChild => 21,
+                FlatStorageReshardingStatus::CatchingUp(_) => 22,
```

Comment:
I don't think this will be needed. CatchingUp should be the same as ready, just a little bit behind.

Reply:
I thought `FlatStorageReshardingStatus::CatchingUp` would be basically the same as `FlatStorageCreationStatus::CatchingUp` 🤔

Reply:
Yeah, agree with Andrea. Resharding catching up should perhaps be similar here to creation catching up

Reply:
All right, I think I'm still missing something but looks good either way.

Reply:
What is the difference between how Ready and CatchingUp work? Is CatchingUp used just to mark Flat Storage as not ready to prevent usage because it would be too slow due to having plenty of deltas?